packages/synthetix-main/contracts/satellites/account/AccountBase.sol

```diff
@@ -0,0 +1,121 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/token/ERC721.sol";
+import "@synthetixio/core-contracts/contracts/utils/AddressUtil.sol";
+import "@synthetixio/core-contracts/contracts/initializable/InitializableMixin.sol";
+import "../../storage/AccountStorage.sol";
+import "../../interfaces/IDelegation.sol";
+
+/// @dev Common utils, errors, and events to be used by any contracts that conform the AcccountModule
+contract AccountBase is ERC721, AccountStorage, IDelegation, InitializableMixin {
+    using SetUtil for SetUtil.AddressSet;
+
+    // ---------------------------------------
+    // Errors
+    // ---------------------------------------
+
+    error NotAuthorized(uint256 account, address addr, Permission permission);
+
+    // ---------------------------------------
+    // Events
+    // ---------------------------------------
+
+    event PermissionGranted(uint256 account, address authorized, Permission permission);
+    event PermissionRevoked(uint256 account, address authorized, Permission permission);
+    event PermissionsSet(uint256 account, address authorized, uint32 packedPermissions);
+    event PermissionsCleared(uint256 account, address authorized);
+
+    // ---------------------------------------
+    // Helpers
+    // ---------------------------------------
+
+    function _isInitialized() internal view override returns (bool) {
+        return _accountStore().initialized;
+    }
+
+    function _getPermissionMask(Permission permission) internal pure returns (uint32) {
+        return uint32(1) << uint32(permission);
+    }
+
+    function _hasPermission(
+        uint256 account,
+        address authorized,
+        Permission permission
+    ) internal view returns (bool) {
+        return _accountStore().delegatedPermissions[account][authorized] & _getPermissionMask(permission) != 0;
+    }
+
+    function _grantPermission(
```

Comment:
what is the purpose of having these extra functions if a user/web ui can just call `_setPermissions` to accomplish same task?

Reply:
I added both because the interface wasn't declared yet to support single call to set all the permissions in one step or single permission granting/revoking operations.
If they are not required I'll delete the single perm operations.